src/components/visualizations/chart/chartOptions/comboChartOptions.ts

```diff
@@ -1,47 +1,59 @@
-// (C) 2007-2018 GoodData Corporation
+// (C) 2007-2019 GoodData Corporation
+import set = require('lodash/set');
+import get = require('lodash/get');
+import cloneDeep = require('lodash/cloneDeep');
+import { Execution, VisualizationObject as VizObject } from '@gooddata/typings';
 import { MEASURES, SECONDARY_MEASURES } from '../../../../constants/bucketNames';
+import { IChartConfig, ISeriesItem } from '../../../../interfaces/Config';
+import { VisualizationTypes } from '../../../../constants/visualizationTypes';
 
-export function getComboChartOptions(
-    config: any,
-    measureGroup: any,
-    series: any,
-    categories: any
-) {
-    const { mdObject } = config;
-    const measureGroupIdentifiers = measureGroup.items.map((item: any) =>
-        item.measureHeaderItem.localIdentifier);
+export const CHART_ORDER = {
+    [VisualizationTypes.COLUMN]: 1,
+    [VisualizationTypes.AREA]: 2,
+    [VisualizationTypes.LINE]: 3
+};
+
+const DEFAULT_COMBO_CHART_TYPES = [
+    VisualizationTypes.COLUMN,
+    VisualizationTypes.LINE
+];
+
+export function getComboChartSeries(
+    config: IChartConfig,
+    measureGroup: Execution.IMeasureGroupHeader['measureGroupHeader'],
+    series: ISeriesItem[]
+): ISeriesItem[] {
+    const updatedSeries = cloneDeep(series);
     const measureBuckets = {};
+    const types = [config.primaryChartType, config.secondaryChartType];
+    const buckets = get(config, ['mdObject', 'buckets'], []);
+    const measureGroupIdentifiers = measureGroup.items.map(
+        (item: Execution.IMeasureHeaderItem) => get(item, ['measureHeaderItem', 'localIdentifier'])
+    );
 
-    if (mdObject) {
-        mdObject.buckets.forEach((bucket: any) => {
-            const bucketItems = bucket.items;
-            const metricIndexes: number[] = [];
-            bucketItems.forEach((item: any) => {
-               if (item.measure) {
-                 const metricIndex = measureGroupIdentifiers.indexOf(item.measure.localIdentifier);
-                 metricIndexes.push(metricIndex);
-               }
-            });
-            measureBuckets[bucket.localIdentifier] = metricIndexes;
-        });
-    }
+    buckets.forEach((bucket: VizObject.IBucket) => {
+        const bucketItems = bucket.items || [];
 
-    if (measureBuckets[MEASURES]) {
-        measureBuckets[MEASURES].forEach((measureIndex: number) => {
-            series[measureIndex].type = 'column';
-        });
-    }
+        measureBuckets[bucket.localIdentifier] = bucketItems.reduce((result: number[], item: VizObject.BucketItem) => {
```

Comment:
Put this logic (bucketItems.reduce((result: number[], item: VizObject.BucketItem) => { ...)
Into separate function name it, it will make code more readable.